File: JavaScript/Co-op/point_to_set.js
//- Function to calculate the average pairwise distance for a point p and a set of points S
function pd_avg(p, S, measure = null, func = null) {
    let res = 0;
    S.forEach(row => {
      res += pd(p, Object.values(row), measure, func);
    });
    return res / S.length;
  }
  
  //- Function to calculate the weighted average pairwise distance for a point p and a set of points S
function pd_avg_w(p, S, weights, measure = null, func = null) {
    let res = 0;
    S.forEach(row => {
      res += pd_w(p, Object.values(row), weights, measure, func);
    });
    return res / S.length;
  }
  
console.log("point_to_set.js loaded");

File: Python/Co-op/measures/point_to_set.py
import point_to_point as p2p
import pandas as pd

#- Function to calculate the average pairwise distance for a point p and a set of points S
def pd_avg(p, S, measure=None, function=None):
    res = 0
    for index, row in S.iterrows():
        res += p2p.pd(p, row.tolist(), measure, function)
    return res/len(S)

#- Function to calculate the weighted average pairwise distance for a point p and a set of points S
def pd_avg_w(p, S, weights, measure=None, function=None):
    res = 0
    for index, row in S.iterrows():
        res += p2p.pd_w(p, row.tolist(), weights, measure, function)
    return res/len(S)

